### PR TITLE
Lpfg 568 add module video

### PR DESCRIPTION
**Pull request by mick-jones**
- Added module template/controller/factory
- Added module tests
- Added modules selector form functionality - page redirects still to be implemented
- Fixed url for edit course -> module

### CODE REVIEW

File: src/applicationContext.ts [hunk 1]
```diff
@@ -25,6 +25,11 @@ import {LearningProvider} from './learning-catalogue/model/learningProvider'
 import {CancellationPolicy} from './learning-catalogue/model/cancellationPolicy'
 import {TermsAndConditions} from './learning-catalogue/model/termsAndConditions'
 import {Course} from './learning-catalogue/model/course'
```
Comment (elxsy) [inline src/applicationContext.ts]: ApplicationContext getting bigger and bigger, can we break this down to Components, please?

File: src/controllers/moduleController.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		this.learningCatalogue = learningCatalogue
+		// this.moduleValidator = moduleValidator
+		this.moduleFactory = moduleFactory
+		this.router = Router()
+
+		this.setRouterPaths()
+	}
+
+	private setRouterPaths() {
+		this.router.param('courseId', async (req, res, next, courseId) => {
```
Comment (elxsy) [inline src/controllers/moduleController.ts]: Your method says it sets Router Paths but actually fetches course from route itself also. Is it possible to extract fetching to a different private method with appropriate name?

File: src/applicationContext.ts (hunk 1 above)
Comment (m24murray) [reply src/applicationContext.ts]: Fixing in later PR

File: src/controllers/moduleController.ts (hunk 2 above)
Comment (m24murray) [reply src/controllers/moduleController.ts]: Fixing in later PR

**Pull request merged**